### PR TITLE
DOCSP-36564 1.4.1 Release Notes

### PR DESCRIPTION
**Pull request by ianf-mongodb**
## DESCRIPTION

Adding 1.4.1 Compass Release Notes.

## STAGING

https://preview-mongodbianfmongodb.gatsbyjs.io/compass/DOCSP-36564/release-notes/#compass-1.42.1

## JIRA

https://jira.mongodb.org/browse/DOCSP-36564

## BUILD LOG

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=65ce6e947111165982046bde

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/release-notes.txt [hunk 1]
```diff
@@ -10,6 +10,27 @@ Release Notes
...
+New Features
+
+- Added vector search index creation to create search indexes 
+  modal :issue:`COMPASS-7302`.
+- Vector search type indexes now display in the search index table 
+  :issue:`COMPASS-7509`.
+
+Bug Fixes
+
+- Improved validation of command line arguments :issue:`COMPASS-7260`.
```
Comment (ianf-mongodb) [inline source/release-notes.txt:27]: I dropped the mention of intercom / active user information in the [Github Release Notes](https://github.com/mongodb-js/compass/releases/tag/v1.42.1) as this seemed like a internal concept. LMK if you disagree.

File: source/release-notes.txt [hunk 2]
```diff
@@ -10,6 +10,27 @@ Release Notes
...
 
+|compass| 1.42.1
+----------------
+
+*Released February 15, 2024*
+
+New Features
+
+- Added vector search index creation to the create search indexes 
+  modal :issue:`COMPASS-7302`.
```
Comment (ltran-mdb2) [inline source/release-notes.txt:21]:
```suggestion
  modal (:issue:`COMPASS-7302`).
```

Comment (ltran-mdb2) [reply source/release-notes.txt:21]: [suggestion] adding parenthesis around the JIRAs to remain consistent with the other release notes on the page

File: source/release-notes.txt [hunk 3]
```diff
@@ -10,6 +10,27 @@ Release Notes
...
+- Vector search type indexes now display in the search index table 
+  :issue:`COMPASS-7509`.
```
Comment (ltran-mdb2) [inline source/release-notes.txt:23]:
```suggestion
  (:issue:`COMPASS-7509`).
```

File: source/release-notes.txt [hunk 4]
```diff
@@ -10,6 +10,27 @@ Release Notes
...
+
+Bug Fixes
+
+- Improved validation of command line arguments :issue:`COMPASS-7260`.
```
Comment (ltran-mdb2) [inline source/release-notes.txt:27]:
```suggestion
- Improved validation of command line arguments (:issue:`COMPASS-7260`).
```

**Review by ltran-mdb2 — APPROVED** [review]
LGTM mod a small suggestion! Thanks @ianf-mongodb !

File: source/release-notes.txt (hunk 2 above)
Comment (ianf-mongodb) [reply source/release-notes.txt:21]: Agree, good catch thanks @ltran-mdb2

**Review by Anemy — APPROVED** [review]

**Pull request merged**